tests/virt_autotest/sriov_pci_passthrough.pm

```diff
@@ -0,0 +1,279 @@
+# SUSE's openQA tests
+ #
+ # Copyright (C) 2019 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+ 
+# Summary: A test on KVM host to pass a SR-IOV NIC VF to guest via libvirt.
+# Maintainer: Julie CAO <JCao@suse.com>
+ 
+use base "virt_feature_test_base";
+use strict;
+use warnings;
+use testapi;
+use set_config_as_glue;
+use utils;
+use virt_utils;
+use xen;
+use virt_autotest::virtual_network_utils qw(save_guest_ip test_network_interface);
+ 
+sub run_test {
+  my $self = shift;
+
+#  my $hypervisor;
+#  if (check_var("HOST_HYPERVISOR", "kvm") || check_var("SYSTEM_ROLE", "kvm") || check_var("HYPERVISOR", "kvm")) {
+#    $hypervisor = "kvm";
+#  }
+#  else {
+#    $hypervisor = "xen";
+#  }
+
+  #get the SR-IOV device BDF and interface 
+  my @host_pfs;
+  @host_pfs = get_sriov_devices ();
+  $self->{test_results}->{"host"}->{"Error: not found any SR-IOV network device in the host!"}->{status} = 'FAILED' if ! $#host_pfs;
+  record_info("Find SR-IOV devices", "@host_pfs");
+  
+  # enable 8 vfs for the SR-IOV device on host
+  my @host_vfs = enable_vf(@host_pfs);
+  $self->{test_results}->{"host"}->{"Error: fail to enable the SR-IOV PF @host_pfs!"}->{status} = 'FAILED' if ! @host_vfs;
+  record_info("VFs enabled", "@host_vfs");
+
+  #generate ssh key on host
+  script_run "rm ~/.ssh/id_rsa*; ssh-keygen -t rsa -f ~/.ssh/id_rsa -P ''";
+
+  foreach my $guest (keys %xen::guests) {
+  my $guest = "sles-15-sp2-64-fv-def-net";
+
+    prepare_guest($guest);
+    
+    #detach one of the vfs from host
+    my %host_vf;
+    $host_vf{bdf} = $host_vfs[int(rand(7))];  #pick up randomly
+    $host_vf{id} = get_device_id($host_vf{bdf});
+    record_info("Detach VF from host", "$host_vf{id}");
+    assert_script_run "virsh nodedev-detach $host_vf{id}";
+
+    #plugin vf to vm via virsh
+    record_info("Pluging vf to vm via libvirt", "guest=$guest \nvf=$host_vf{id}");
+    my %vf_vm = plugin_device_kvm($guest, $host_vf{id});
+    record_info("VF plugg'ed in", "bdf=$vf_vm{bdf}, mac=$vf_vm{mac}");
+  
+    #check the networking of the plugged interface
+    my $gateway = script_output "ip r s | grep 'default via' | cut -d ' ' -f3";
+    set_var("SRIOV_PCI_PASSSHTROUGH", 1);   #to differenciate virtual network tests
+    test_network_interface($guest, gate => $gateway, mac => $vf_vm{mac}, net => 'br123');   #use br123 as ssh connection
+  
+    #reboot
```

Comment:
There should be
# reboot guest

Reply:
Yes, change it.

Reply:
Great, should be marked as resolved here.